CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ValueSerializer.swift

```diff
@@ -0,0 +1,156 @@
+            let range = start..<end
+            return Int(value!.substring(with: range)) as? T
+            
+        } else if value!.hasPrefix("F") {
+            let start = value!.index(value!.startIndex, offsetBy: 1)
+            let end = value!.index(value!.endIndex, offsetBy: 0)
+            let range = start..<end
+            return Float(value!.substring(with: range)) as? T
+           
+        } else if value!.hasPrefix("D") {
+            let start = value!.index(value!.startIndex, offsetBy: 1)
+            let end = value!.index(value!.endIndex, offsetBy: 0)
+            let range = start..<end
+            return Double(value!.substring(with: range)) as? T
+        } else if value!.hasPrefix("L") {
+            let start = value!.index(value!.startIndex, offsetBy: 1)
+            let end = value!.index(value!.endIndex, offsetBy: 0)
+            let range = start..<end
+            return Int64(value!.substring(with: range)) as? T
+        } else if value!.hasPrefix("#") {
+                if (value?.range(of:".")) != nil {
+                    return Double(value!) as? T
+                } else {
+                    return Int(value!) as? T
+                }
+        } else if (value == "true") {
+            return true as? T
+        } else if (value == "false") {
+            return false as? T
+        } else if (value!.hasPrefix("{")) {
+            let data: Data = value!.data(using: String.Encoding.utf8)!
+            var stringMap = [String: Any]()
+            do {
+                stringMap = try JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions.mutableContainers) as! Dictionary<String, Any>
+            } catch {
+                return nil
+            }
+            var map: Dictionary = [String: Any]()
+            
+            for map_param in stringMap {
+                let key = map_param.key
+                if let key_value = deserialize(value: map_param.value as? String, memory: memory) as Any? {
+                     map[key] = key_value
+                }
+                else{
+                    map[key] = NSNull()
+                }
+            }
+            return map as? T
+        } else if (value!.hasPrefix("[")) {
+            let data: Data = value!.data(using: String.Encoding.utf8)!
+            var stringList = [String]()
+            do {
+                stringList = try JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions.mutableContainers) as! [String]
+            } catch {
+                return nil
+            }
+            var list = [Any]()
+            
+            for string in stringList {
+                 let object: Any? = deserialize(value: string, memory: memory)
+                 list.append(object ?? NSNull.init())
+            }
+            return list as? T
+        } else {
+            return "Invalid value type \(String(describing: value))" as? T
```

Comment:
I remember once docs created by cbl is adding  in document with invalid value type data . Not sure if that is fixed now or still happening. If not , we should fix it

Reply:
Since tests are passing, I am assuming it is fixed. Let me know if that is not the case with iOS.

Reply:
Tests will pass, but the content shows 'invalid value type' description. Can you run test by cbl and see if does not create docs with contents 'invalid value type"

Reply:
I can change it to
```
print("Invalid value type \(String(describing: value))")
return nil
```
What are we expecting the behavior to be here?

Reply:
I have also tested out the above change with a replication test case